iroh-bytes/src/store/file.rs

```diff
@@ -0,0 +1,2483 @@
+    /// Complete data files
+    pub complete: PathBuf,
+    /// Partial data files
+    pub partial: PathBuf,
+    /// Metadata files such as the tags table
+    pub meta: PathBuf,
+}
+
+#[cfg(test)]
+#[derive(Debug)]
+pub(crate) struct EntryStateResponse {
+    mem: Option<BaoFileHandle>,
+    db: Option<EntryState<Bytes>>,
+}
+
+///
+#[derive(Debug, Clone)]
+pub struct Store(Arc<StoreInner>);
+
+impl Store {
+    /// Load or create a new store.
+    pub async fn load(root: impl AsRef<Path>) -> io::Result<Self> {
+        let path = root.as_ref();
+        let db_path = path.join("meta").join("blobs.db");
+        let options = Options {
+            path: PathOptions::new(path),
+            inline: Default::default(),
+        };
+        Self::new(db_path, options).await
+    }
+
+    async fn new(path: PathBuf, options: Options) -> io::Result<Self> {
+        // spawn_blocking because StoreInner::new creates directories
+        let rt = tokio::runtime::Handle::try_current()
+            .map_err(|_| io::Error::new(io::ErrorKind::Other, "no tokio runtime"))?;
+        let inner =
+            tokio::task::spawn_blocking(move || StoreInner::new_sync(path, options, rt)).await??;
+        Ok(Self(Arc::new(inner)))
+    }
+
+    ///
+    pub async fn update_inline_options(
+        &self,
+        inline_options: InlineOptions,
+        reapply: bool,
+    ) -> io::Result<()> {
+        Ok(self
+            .0
+            .update_inline_options(inline_options, reapply)
+            .await?)
+    }
+
+    ///
+    pub async fn dump(&self) -> io::Result<()> {
+        Ok(self.0.dump().await?)
+    }
+
+    ///
+    pub async fn sync(&self) -> io::Result<()> {
+        Ok(self.0.sync().await?)
+    }
+
+    /// Import from a v0 or v1 flat store, for backwards compatibility.
+    pub async fn import_flat_store(&self, paths: FlatStorePaths) -> io::Result<bool> {
+        Ok(self.0.import_flat_store(paths).await?)
+    }
+
+    /// Get the complete state of an entry, both in memory and in redb.
+    #[cfg(test)]
+    pub(crate) async fn entry_state(&self, hash: Hash) -> io::Result<EntryStateResponse> {
+        Ok(self.0.entry_state(hash).await?)
+    }
+
+    #[cfg(test)]
+    pub fn owned_data_path(&self, hash: &Hash) -> PathBuf {
+        self.0.path_options.owned_data_path(hash)
+    }
+
+    #[cfg(test)]
+    pub fn owned_outboard_path(&self, hash: &Hash) -> PathBuf {
+        self.0.path_options.owned_outboard_path(hash)
+    }
+}
+
+#[derive(Debug)]
+struct StoreInner {
+    tx: flume::Sender<ActorMessage>,
+    temp: Arc<RwLock<TempCounterMap>>,
+    handle: Option<std::thread::JoinHandle<()>>,
+    path_options: Arc<PathOptions>,
+}
+
+impl LivenessTracker for RwLock<TempCounterMap> {
+    fn on_clone(&self, content: &HashAndFormat) {
+        self.write().unwrap().inc(content);
+    }
+
+    fn on_drop(&self, content: &HashAndFormat) {
+        self.write().unwrap().dec(content);
+    }
+}
+
+impl StoreInner {
+    pub fn new_sync(
```

Comment:
why are all the `StoreInner` methods public?

Reply:
They used to be directly on Store.